### DISCUSSION TITLE
Getting embeddings

### DISCUSSION THREAD

**Discussion by johnyquest7** [KeremTurgutlu/self_supervised#22] [Q&A] [answered]
thank you for creatng this awesome repository.
How do I get embeddings out of the SwAV model for one image?

**Comment by KeremTurgutlu** [comment, marked as answer]
Assuming you trained with a learner such as:
```
dls = get_dls(resize, bs)
model = create_swav_model(arch=xresnet34, n_in=3, pretrained=False)
learn = Learner(dls, model, SWAVLoss(),
                cbs=[SWAV(crop_sizes=[size,96], 
                          num_crops=[2,6],
                          min_scales=[0.25,0.2],
                          max_scales=[1.0,0.35]),
                     TerminateOnNaNCallback()])
```

You can simply use `learn.model.encoder` and feed input image to extract it's embedding. Don't forget to process the image with the same transforms you used during training, like normalization.

If this doesn't help, please feel free to share link to a notebook that you have so that we can debug it together.

**Reply by KeremTurgutlu** [reply]
I am currently working on `v1.0.0`, and I am planning on putting embedding extraction as an example notebook :)

**Reply by johnyquest7** [reply]
Awesome

**Reply by KeremTurgutlu** [reply]
Until the official release of new version you can checkout this [example notebook](https://github.com/KeremTurgutlu/self_supervised/blob/f0e1f313a6a46682ba36c4e5e676d00e528084a4/examples/10%20-%20embedding_extraction.ipynb).

**Reply by johnyquest7** [reply]
Thank you for the detailed example. Will have to try it out. 
It was very easy to train a model with your code. I am seeing pretty good results and it seems to be really fast!
Saw your to-do list - waiting for - Medical domain experiments/examples

**Reply by KeremTurgutlu** [reply]
> It was very easy to train a model with your code. I am seeing pretty good results and it seems to be really fast!

This comment makes me very happy :) The new version I am working on is even faster and better, you can checkout few experiments here on [imagewang benchmark](https://github.com/fastai/imagenette/pull/44)

> Saw your to-do list - waiting for - Medical domain experiments/examples

My plan is to start with chest X-rays and try it out on few downstream tasks such as Kaggle competitions or other leaderboards. It will be interesting and hard to find out which augmentations are going to work for medical data. Let me know if you already have an idea :)

**Reply by johnyquest7** [reply]
CLAHE worked better for one of my medical projects. Not sure whether it was a random finding. https://albumentations.ai/docs/api_reference/augmentations/transforms/#albumentations.augmentations.transforms.CLAHE
